core/common/src/main/java/alluxio/underfs/UnderFileStatus.java

```diff
@@ -66,6 +66,9 @@ public String toString() {
    * @return array of file or directory names
    */
   public static String[] toListingResult(UnderFileStatus[] children) {
+    if (children == null) {
+      return null;
```

Comment:
A null return is used to detect a non-existent directory; returning an empty array means we cannot distinguish w/ an empty directory